src/android/com/microsoftopentechnologies/intellij/helpers/storage/TableFileEditor.java

```diff
@@ -0,0 +1,484 @@
+
+                            AzureSDKManagerImpl.getManager().deleteTableEntity(storageAccount, selectedEntities[i]);
+                        }
+
+                        ApplicationManager.getApplication().invokeLater(new Runnable() {
+                            @Override
+                            public void run() {
+                                tableEntities.removeAll(Arrays.asList(selectedEntities));
+
+                                refreshGrid();
+                            }
+                        });
+                    }
+                } catch(AzureCmdException ex) {
+                    UIHelper.showException("Error deleting entities", ex, "Service Explorer", false, true);
+                }
+            }
+        });
+    }
+
+    private TableEntity[] getSelectedEntities() {
+        if(tableEntities == null) {
+            return null;
+        }
+
+        int partitionIdIndex = -1;
+        int rowIdIndex = -1;
+
+        for(int i = 0; i < entitiesTable.getColumnCount(); i++) {
+            String columnName = entitiesTable.getColumnName(i);
+            if(columnName.equals(PARTITION_KEY)){
+                partitionIdIndex = i;
+            }
+            if(columnName.equals(ROW_KEY)){
+                rowIdIndex = i;
+            }
+        }
+
+        ArrayList<TableEntity> selectedEntities = new ArrayList<TableEntity>();
+
+        for (int i : entitiesTable.getSelectedRows()) {
+            for (TableEntity tableEntity : tableEntities) {
+                String partitionValue = entitiesTable.getValueAt(i, partitionIdIndex).toString();
+                String rowIdValue = entitiesTable.getValueAt(i, rowIdIndex).toString();
+                if(tableEntity.getPartitionKey().equals(partitionValue)
+                        && tableEntity.getRowKey().equals(rowIdValue)){
+                    selectedEntities.add(tableEntity);
+                }
+            }
+        }
+
+        return selectedEntities.toArray(new TableEntity[selectedEntities.size()]);
+    }
+
+    @NotNull
+    public static String getFormattedProperty(@NotNull TableEntity.Property property) {
+        try {
+            switch (property.getType()) {
+                case Boolean:
+                    return property.getValueAsBoolean().toString();
+                case DateTime:
+                    return new SimpleDateFormat().format(property.getValueAsCalendar().getTime());
+                case Double:
+                    return property.getValueAsDouble().toString();
+                case Integer:
+                    return property.getValueAsInteger().toString();
+                case Long:
+                    return property.getValueAsLong().toString();
+                case Uuid:
+                    return property.getValueAsUuid().toString();
+                case String:
+                    return property.getValueAsString();
+            }
+        } catch (AzureCmdException ignored) {}
+
+        return "";
+    }
```

Comment:
Maybe we can simplify this to something like the following after adding a `getValue` getter to the `Property` class that returns the underlying the `value` field?

    @NotNull
    public static String getFormattedProperty(@NotNull TableEntity.Property property) {
        try {
            switch (property.getType()) {
                case Boolean:
                case Double:
                case Integer:
                case Long:
                case Uuid:
                case String:
                    return property.getValue().toString();
                case DateTime:
                    return new SimpleDateFormat().format(property.getValueAsCalendar().getTime());
            }
        } catch (AzureCmdException ignored) {}

        return "";
    }